src/Maker/AdminMaker.php

```diff
@@ -0,0 +1,249 @@
+use Sonata\AdminBundle\Model\ModelManagerInterface;
+use Symfony\Bundle\MakerBundle\ConsoleStyle;
+use Symfony\Bundle\MakerBundle\DependencyBuilder;
+use Symfony\Bundle\MakerBundle\Generator;
+use Symfony\Bundle\MakerBundle\InputConfiguration;
+use Symfony\Bundle\MakerBundle\Maker\AbstractMaker;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\DependencyInjection\Container;
+
+/**
+ * @author Gaurav Singh Faudjdar <faujdar@gmail.com>
+ */
+class AdminMaker extends AbstractMaker
+{
+    /**
+     * @var string
+     */
+    private $projectDirectory;
+
+    /**
+     * @var ModelManagerInterface[]
+     */
+    private $availableModelManagers;
+
+    public function __construct($projectDirectory, array $modelManagers = [])
+    {
+        $this->projectDirectory = $projectDirectory;
+        $this->availableModelManagers = $modelManagers;
+    }
+
+    /**
+     * Return the command name for your maker (e.g. make:report).
+     *
+     * @return string
+     */
+    public static function getCommandName(): string
+    {
+        return 'make:sonata:admin';
+    }
+
+    /**
+     * Configure the command: set description, input arguments, options, etc.
+     *
+     * By default, all arguments will be asked interactively. If you want
+     * to avoid that, use the $inputConfig->setArgumentAsNonInteractive() method.
+     *
+     * @param Command            $command
+     * @param InputConfiguration $inputConfig
+     */
+    public function configureCommand(Command $command, InputConfiguration $inputConfig)
+    {
+        $command
+            ->setDescription('Generates an admin class based on the given model class')
+            ->addArgument('model', InputArgument::REQUIRED, 'The fully qualified model class')
+            ->addOption('admin', 'a', InputOption::VALUE_OPTIONAL, 'The admin class basename')
+            ->addOption('controller', 'c', InputOption::VALUE_OPTIONAL, 'The controller class basename')
+            ->addOption('manager', 'm', InputOption::VALUE_OPTIONAL, 'The model manager type')
+            ->addOption('services', 'y', InputOption::VALUE_OPTIONAL, 'The services YAML file', 'services.yaml')
+            ->addOption('id', 'i', InputOption::VALUE_OPTIONAL, 'The admin service ID');
+
+        $inputConfig->setArgumentAsNonInteractive('model');
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function interact(InputInterface $input, ConsoleStyle $io, Command $command)
+    {
+        $io->section('Welcome to the Sonata admin maker');
+        $modelClass = $io->ask(
+            'The fully qualified model class',
+            $input->getArgument('model'),
+            'Sonata\AdminBundle\Command\Validators::validateClass'
+        );
+        $modelClassBasename = current(array_slice(explode('\\', $modelClass), -1));
```

Comment:
https://stackoverflow.com/questions/19901850/how-do-i-get-an-objects-unqualified-short-class-name?utm_medium=organic&utm_source=google_rich_qa&utm_campaign=google_rich_qa

Reply:
Please use reflection here, it's faster and easier to understand